### ISSUE TITLE
Loop logger

### ISSUE THREAD

**Issue by meodemsao** [prisma/graphql-middleware#22]
I using with logger middleware and console.log loop many in console ( i don't query or call graphql )

![screen shot 2018-08-08 at 4 09 06 pm](https://user-images.githubusercontent.com/570103/43827954-6b48bfd2-9b25-11e8-9c00-bc9bea16d674.png)

![screen shot 2018-08-08 at 4 07 53 pm](https://user-images.githubusercontent.com/570103/43827913-464c7dfe-9b25-11e8-843d-f2a275e251a6.png)

**Comment by maticzav** [comment]
Hey @meodemsao 👋

What’s the query that you are using?

**Comment by meodemsao** [comment]
@maticzav i using prisma query, when i running server logger is looping, i don't have call graphql

**Comment by meodemsao** [comment]
@maticzav have any idea for this issues ?

**Comment by maticzav** [comment]
Hey, I am trying to figure out why this is happening. Are you making any calls?

**Comment by maticzav** [comment]
Hey @meodemsao,

I am unable to reproduce the bug. Could you verify that it's still happening? If so, could you try making a reproduction repository or copy your code to [glitch 🐟 ](https://glitch.com) so I can help you out?

**Comment by meodemsao** [comment]
@maticzav 

my app in glitch

https://responsible-mushroom.glitch.me
console.log appear when i start call http://localhost:4000 -> i think maybe graphql-playground call to graphql-yoga many time

**Comment by maticzav** [comment]
@meodemsao where can I see the logs from playground?

**Comment by meodemsao** [comment]
@maticzav you can see log in terminal when you start using playground on http://localhost:4000

**Comment by meodemsao** [comment]
@maticzav i need sure about this issues because if i using another logger like https://github.com/winstonjs/winston maybe log too much more

**Comment by maticzav** [comment]
@meodemsao the link you sent above only points to the playground; not the actual code itself. Can you provide a reprodiction code that is not an image? Otherwise, it is extremely difficult for me to help you.

**Comment by meodemsao** [comment]
@maticzav i only run prisma + graphql-middleware and have this issues
i think you only setup graphql-middleware and prisma you can see this issues in terminal

**Comment by moritzmorgenroth** [comment]
Hi @maticzav, we are running into similar behavior using the middlewhere in one of our microservices. To reproduce, please check out https://github.com/twigbit/vos_objectservice, commit f506db6c391e4f201a81fe5340582f6e450990ba and run `yarn dev`. You should receive console output like so ```first middlewhere called
first middlewhere called
first middlewhere called
first middlewhere called
first middlewhere called
second middlewhere called
second middlewhere called
second middlewhere called
second middlewhere called
second middlewhere called
first middlewhere called
first middlewhere called
first middlewhere called
first middlewhere called
first middlewhere called
first middlewhere called
first middlewhere called``` Let me know it you can reproduce. Thanks!

**Comment by meodemsao** [comment]
@maticzav with prisma how we can limit this ?

**Comment by meodemsao** [comment]
@maticzav  how we distinguish between schema and request ?

**Comment by maticzav** [comment]
@meodemsao I am not sure I understand your question. Could you elaborate?

**Comment by moritzmorgenroth** [comment]
@maticzav Thanks a lot for clearing this up! A reference in the docs would be great since for many use cases this is probably the desired behavior. From my point of view, it might even be nice to make this behavior default and give an option to explicitly act before introspection queries as well.

**Comment by yhjor** [comment]
@maticzav I have the same error and there is no way to detect which lifecycle it is logging to. It always runs more than the actual call of the resolvers.

**Comment by maticzav** [comment]
@moritzmorgenroth I like your idea. Nevertheless, I believe this is the default behaviour. GraphQL by default includes `__schema` queries in `Query`. I think your approach works well in simple use cases but lacks scalability and consistency across GraphQL Middleware.

@yhjor I cannot reproduce the bug.

---

We have clarified the cause of the initial problem and identified it as expected behaviour, not a bug. I would like to point out that this library has been actively used internally before its release to make sure there's no unexpected behaviour. From experience, I have gathered since the release; I can observe that majority of problems erupt from a poor understanding of GraphQL concepts. In light of this, I'll close this issue. I'll still happily explain and respond to every comment below which will describe `expected vs actual` result and include the `schema` for reproduction.
